tiledb/sm/misc/mgc_dict.cc

```diff
@@ -37,45 +39,30 @@
 namespace tiledb {
 namespace sm {
 
-static const char magic_mgc_compressed_bytes[] = {
-#include "magic_mgc_gzipped.bin"
-};
+const tiledb::sm::ByteVecValue prepare_data() {
+  tiledb::sm::ByteVecValue expanded_buffer(magic_mgc_decompressed_size);
```

Comment:
We shouldn't create new uses of  `ByteVecValue`. Even `std::vector` would be better. Since `magic_mgc_decompressed_size` is `constexpr`, though, it would be even better to use `std::array` as the innermost instance. Putting that inside `shared_ptr` would be adequate if we want to continue allocating the structure every time we use it.

I don't know the uncompressed size, but there's certainly a tradeoff involved in decompressing each time vs. decompressing once and, correspondingly, using an allocated structure vs. a static one. That tradeoff needs documentation. It's less important at this stage to optimize the tradeoff than to (1) state that there is one and (2) outline its nature. This documentation should be in the header, not here with the code.

* We are resolving the tradeoff by decompressing each time. That needs to be stated clearly in documentary text form, not buried in code-specific comments.
* The documentation should at least state some sizes for a particular named version of the compressed and uncompressed sizes. It should also state that we have not yet written code to measure the decompression time that would better inform the tradeoff.

Reply:
I will move away from `ByteVecValue`. Because the uncompressed file has a size of 6.7MB, I fear that an `std::array` will overflow the stack. I could use a unique pointer to an array but I don't think it would be much more efficient than a simple vector.

Regarding the tradeoffs, I believe 6.7MB is a size too big to repeatedly decompress and allocate for each use, but not too big to keep in memory for reuse after the first operation. And as stated above it is definitely too big to put in the stack. I will add documentation explaining that.

Reply:
> We are resolving the tradeoff by decompressing each time. That needs to be stated clearly in documentary text form, not buried in code-specific comments.

We do not decompress each time, if I correctly understood that this is what you understood. `prepare_data()` is private to this compilation unit, and gets called by `expanded_buffer()` at the bottom of the file to initialize a `static` field. I just validated it by running `unit_mgc_dict` with a breakpoint inside `prepare_data()` and observing that the breakpoint is hit only once.